src/main/drivers/max7456.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <stdint.h>
+
+#include "common/printf.h"
+#include "platform.h"
+#include "version.h"
+
+#ifdef USE_MAX7456
+
+#include "drivers/bus_spi.h"
+#include "drivers/system.h"
+
+#include "max7456.h"
+
+#define DISABLE_MAX7456       GPIO_SetBits(MAX7456_CS_GPIO,   MAX7456_CS_PIN)
+#define ENABLE_MAX7456        GPIO_ResetBits(MAX7456_CS_GPIO, MAX7456_CS_PIN)
+
+
+uint16_t max_screen_size;
+uint8_t  video_signal_type = 0;
+uint8_t  max7456_lock = 0;
+char screen[480];
```

Comment:
/** PAL or NTSC, value is number of chars total */
#define VIDEO_MODE_PIXELS_NTSC 480
#define VIDEO_MODE_PIXELS_PAL 390